src/seq/seq.jl

```diff
@@ -15,15 +16,69 @@ export Nucleotide, DNANucleotide, RNANucleotide,
        eachkmer, NucleotideCounts, Kmer, DNAKmer, RNAKmer, dnakmer, rnakmer,
        kmer, AminoAcid, AminoAcidSequence, @aa_str, translate, ncbi_trans_table,
        AA_A, AA_R, AA_N, AA_D, AA_C, AA_Q, AA_E, AA_G, AA_H, AA_I, AA_L,
-       AA_K, AA_M, AA_F, AA_P, AA_S, AA_T, AA_W, AA_Y, AA_V, AA_X
+       AA_K, AA_M, AA_F, AA_P, AA_S, AA_T, AA_W, AA_Y, AA_V, AA_X,
+       FASTA, FASTQ
+
+
+abstract FileFormat
+immutable FASTA <: FileFormat end
+immutable FASTQ <: FileFormat end
+
+
+abstract Sequence
 
 
 include("nucleotide.jl")
 include("aminoacid.jl")
 include("geneticcode.jl")
 include("util.jl")
+include("alphabet.jl")
+include("quality.jl")
+
+
+# A sequence record is a named sequence with attached metadata.
+@doc """
+`SeqRecord{S, T}` is a type holding a named sequence of type `S`, along with
+some arbitrary metadata of type `T`.
+""" ->
+type SeqRecord{S, T}
+    name::String
+    seq::S
+    metadata::T
+
+    function SeqRecord(name, seq::S, metadata::T)
```

Comment:
Since this is the default constructor, it could be omitted as long as `SeqRecord()` (line 53) is defined outside the type definition.